src/BaseFactory.php

```diff
@@ -107,4 +106,37 @@ protected function getFactoryFromClassName(string $className): FactoryInterface
 
         return new $factoryClass($this->faker);
     }
+
+    private function buildRelationsForModel(Model $model, string $creationType): Model
+    {
+        foreach ($this->relatedModelFactories as $relationshipName => $factories) {
+            $relation = $model->{$relationshipName}();
+
+            if (method_exists($relation, 'saveMany')) {
+                $relatedModels = $factories->map->make();
+                $model->setRelation($relationshipName, $relatedModels);
+
+                if ($creationType === 'create') {
+                    $relation->saveMany($relatedModels);
+                }
+
+                continue;
+            }
+
+            if (method_exists($relation, 'associate')) {
+                $relatedModels = $factories->map->create();
```

Comment:
If we check the creation type and use "make" instead " create" when needed, the code works. (including the change I suggested for the test)

When we have the tests working (without any models being created) we can think of a nicer way to implement this. As you mentioned it gets quite messy here.

Reply:
OK.